src/dataflow-types/types.rs

```diff
@@ -393,6 +397,8 @@ pub struct ProtobufEncoding {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct SourceDesc {
     pub connector: SourceConnector,
+    /// Optionally, filtering and projection to optionally apply.
+    pub operators: Option<LinearOperator>,
```

Comment:
Why does this need to be an `Option` ? Is there any difference in semantics between `None` and

```
Some(LinearOperator {
    predicates: Vec::new(),
    projection: (0..arity).collect(),
})
```
?

Reply:
The main difference was that it was hard to determine `arity` at various moments where you might want to swap out `operators`, or at moments where you needed to determine if this was a no-op operator. I could believe that could be worked around, but the option also seemed to convey the intent better.

Reply:
Maybe you could make the projection be the complement of the projection instead (i.e. the set of fields that *aren't* demanded). Then `LinearOperator` could just derive default. Not necessary though, do whatever you think is best!